internal/services/datafactory/data_factory_linked_service_sftp_resource.go

```diff
@@ -103,31 +121,43 @@ func resourceDataFactoryLinkedServiceSFTP() *pluginsdk.Resource {
 				},
 			},
 
-			"skip_host_key_validation": {
-				Type:     pluginsdk.TypeBool,
-				Optional: true,
+			"private_key_passphrase": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				Description:   "Specify the password to decrypt the private key if the key file or the key content is password protected",
+				ConflictsWith: []string{"password"},
 			},
 
-			"host_key_fingerprint": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
+			"password": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				ValidateFunc:  validation.StringIsNotEmpty,
+				ConflictsWith: []string{"private_key_content", "private_key_path", "private_key_passphrase"},
+				AtLeastOneOf:  []string{"password", "private_key_content", "private_key_path"},
 			},
 
-			"annotations": {
-				Type:     pluginsdk.TypeList,
-				Optional: true,
-				Elem: &pluginsdk.Schema{
-					Type: pluginsdk.TypeString,
-				},
+			"private_key_content": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				Description:   "SSH private key content in OpenSSH format",
+				ValidateFunc:  validate.SSHPrivateKey,
+				ConflictsWith: []string{"private_key_path", "password"},
+				AtLeastOneOf:  []string{"password", "private_key_content", "private_key_path"},
 			},
 
-			"additional_properties": {
-				Type:     pluginsdk.TypeMap,
+			"private_key_path": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				Description:   "Specify the absolute path to the private key file that the integration runtime can access. This applies only when using a self-hosted integration runtime as opposed to the default Azure-hosted runtime, as indicated by providing a value for `integration_runtime_name`.",
+				ConflictsWith: []string{"private_key_content", "password"},
+				AtLeastOneOf:  []string{"password", "private_key_content", "private_key_path"},
+			},
```

Comment:
Sorry to pop in here unannounced suddenly, I've been trying to not interfere and let you both run with the PR and review, mulling this over though I remembered the [`file`](https://developer.hashicorp.com/terraform/language/functions/file) function in Terraform and suggest that we don't expose this since the file function used in conjunction with the `private_key_content` property would render this property redundant.

Reply:
I think I understand your point, however I want to make sure we have the same understanding; the `private_key_path` is from what i understand most likely **not** a path on the host running Terraform, but rather only applicable if the user is self-hosting their "data factory integration runtime" (whatever that is), and is storing the SSH key on that host. In that case, I would think the `private_key_path` is still useful, no?
And please don't hold back whenever you have feedback, consider this a standing invite!

Reply:
I spun up a self hosted integration runtime and connected it to a data factory, then tried making a SFTP linked service and specified a key path, confirmed that it's a path on the runtime host
<img width="277" alt="image" src="https://github.com/user-attachments/assets/4ae7a1a4-5c3e-414e-8181-d41e2129ee13" />

Reply:
If it's the key path for a remote host, does it still make sense to generate a key pair and write the file locally for the test? https://github.com/hashicorp/terraform-provider-azurerm/blob/4af159bc6c7a7313a1f4f68d1b782ff0d98c4643/internal/services/datafactory/data_factory_linked_service_sftp_resource_test.go#L60-L72

Reply:
It doesn't make sense, I will remove it

Reply:
Done!